### PR TITLE
Enable the image template to fetch values from env

### PR DESCRIPTION
**Pull request by markusthoemmes**
Yeah, this looks a little weird, I know.

My use-case is this: Our CI system wants me to define clear dependencies on images it builds (the images are not built via `ko` but via a mechanism in said CI system). The CI system then publishes the built images pullref (sha based) into an environment variable of my choice.

By allowing the functions in this PR, this enables me to fetch these pullrefs from env via a template like

```
{{with $name := .Name}}{{ fromEnv (toEnvKey (printf "KNATIVE_SERVING_TEST_%s" $name)) }}{{end}}
```

:tada: 

/assign @dprotaso @n3wscott

### CODE REVIEW

File: test/e2e_flags.go [hunk 1]
```diff
@@ -68,7 +70,13 @@ func SetupLoggingFlags() {
 
 // ImagePath is a helper function to transform an image name into an image reference that can be pulled.
 func ImagePath(name string) string {
-	tpl, err := template.New("image").Parse(Flags.ImageTemplate)
+	tpl, err := template.New("image").Funcs(template.FuncMap{
+		"toEnvKey": func(in string) string {
+			underscores := strings.ReplaceAll(in, "-", "_")
```
Comment (julz) [inline test/e2e_flags.go:75]: just to ask the question: do we actually need `toEnvKey`/`fromEnv`? Another approach would be for the template to be (though I guess that ends up a pretty ugly template string that you have to build somehow so 🤷):

~~~~
{{with .Name }}
{{if eq . "readiness"}}docker.io/flibble/blobble{{end}}
{{if eq . "varlog"}}docker.io/flibble/bloop{{end}}
..
~~~~

Im wondering a bit if it's worth merging toEnvKey and fromEnv into something like `fromPrefixedEnv("KNATIVE_SERVING_TEST_", name)` since I find it a bit hard to follow the flow with multiple template functions calling each other

File: test/e2e_flags.go [hunk 2]
```diff
@@ -68,7 +70,13 @@ func SetupLoggingFlags() {
...
+			return strings.ToUpper(underscores)
+		},
+		"fromEnv": os.Getenv,
```
Comment (julz) [inline test/e2e_flags.go:78]: could use `os.ExpandEnv` here which would avoid needing the printf in the template

**Comment by markusthoemmes** [comment]
Thanks @julz for the suggestions. I actually don't need this anymore.
